src/Raven.Server/Documents/Replication/ReplicationLoader.cs

```diff
@@ -391,35 +371,37 @@ public void HandleDatabaseRecordChange()
             var newRecord = LoadDatabaseRecord();
 
             if (newRecord == null)
+            {
+                DropAllOutgoingConnections();
+                MyDatabaseRecord = null;
                 return;
+            }
 
-            var connectionChanged = ReplicationTopology.MyConnectionChanged(newRecord.Topology, Server.NodeTag, Database.Name);
+            var connectionChanged = ReplicationTopology.FindConnectionChanges(newRecord.Topology, Server.NodeTag, Database.Name);
             var conflictSolverChanged = MyDatabaseRecord.ConflictSolverConfig.ConflictResolutionChanged(newRecord.ConflictSolverConfig);
 
             MyDatabaseRecord = newRecord;
 
-            if (connectionChanged)
+            if (connectionChanged.nodesToRemove.Count > 0)
             {
+                // remove old connections
                 if (_log.IsInfoEnabled)
-                    _log.Info("Replication connection were changed. Starting and stopping outgoing replication threads.");
+                    _log.Info("Stopping obselete outgoing replication threads.");
```

Comment:
Better to say more about it, such as what threads you are stopping and their count.